### ISSUE TITLE
Cannot create table on custom schema in distributed mode

### ISSUE THREAD

**Issue by sunng87** [GreptimeTeam/greptimedb#867]
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Frontend, Meta

### What happened?

In distributed mode, tables are always created on default schema, `public`, no matter what schema you are using.

It happens on pg, mysql and http interfaces. 

### What operating system did you use?

any

### How can we reproduce the bug?

```bash
1. log on mysql, `use` a custom schema
2. create table `testtable`
3. show tables, expect `testtable`, result: none
4. switch to default schema `public`
5. show tables, expect none, result: `testtable`
```

**Comment by sunng87** [comment]
Possible root cause:

`table_idents_to_full_name`  in `src/sql/src/statements.rs` tries to resolve catalog and schema from fully qualified table name without information from `QueryContext`.

**Comment by sunng87** [comment]
features may not work because of  `table_idents_to_full_name`:

- drop table
- alter table

**Comment by sunng87** [comment]
Fixed in #922
